src/ax/marshaling/message_test.go

```diff
@@ -32,49 +33,102 @@ var _ = Describe("MarshalMessage", func() {
 })
 
 var _ = Describe("UnmarshalMessage", func() {
+
 	message := &messagetest.Message{
 		Value: "<value>",
 	}
-	data, err := proto.Marshal(message)
+
+	nonaxmsg := &messagetest.NonAxMessage{}
+
+	pbdata, err := proto.Marshal(message)
 	if err != nil {
 		panic(err)
 	}
 
-	It("unmarshals the message using the protocol specified in the content-type", func() {
-		m, err := UnmarshalMessage(
-			"application/vnd.google.protobuf; proto=ax.internal.messagetest.Message",
-			data,
-		)
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(proto.Equal(m, message)).To(BeTrue())
-	})
-
-	It("returns an error if the content-type is invalid", func() {
-		_, err := UnmarshalMessage("", data)
-		Expect(err).Should(HaveOccurred())
-	})
+	_, jsondata, err := MarshalJSON(message)
+	if err != nil {
+		panic(err)
+	}
 
-	It("returns an error if the content-type is not supported", func() {
-		_, err := UnmarshalMessage("application/x-unknown", data)
-		Expect(err).Should(HaveOccurred())
-	})
+	nonaxmsgpbdata, err := proto.Marshal(nonaxmsg)
+	if err != nil {
+		panic(err)
+	}
 
-	It("returns an error if an error occurs unmarshaling the protocol buffers message", func() {
-		_, err := UnmarshalMessage(
-			"application/vnd.google.protobuf; proto=ax.internal.messagetest.Unknown", // note unknown message type
-			data,
-		)
-		Expect(err).Should(HaveOccurred())
-	})
+	_, nonaxmsgjsondata, err := MarshalJSON(nonaxmsg)
+	if err != nil {
+		panic(err)
+	}
 
-	It("returns an error if the buffer contains a protocol buffer message that is not an ax.Message", func() {
-		data, err := proto.Marshal(&messagetest.NonAxMessage{})
-		Expect(err).ShouldNot(HaveOccurred())
+	DescribeTable(
+		"unmarshals the message using the protocol specified in the content-type",
+		func(ct string, data []byte, expected *messagetest.Message) {
+			m, err := UnmarshalMessage(
+				ct,
+				data,
+			)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(proto.Equal(m, message)).To(BeTrue())
+		},
+		Entry(
+			"unmarshals the message using the protocol specified in the content-type",
+			"application/vnd.google.protobuf; proto=ax.internal.messagetest.Message",
+			pbdata,
+			message,
+		),
+		Entry(
+			"unmarshals the message using JSON specified in the content-type",
+			"application/json; proto=ax.internal.messagetest.Message",
+			jsondata,
+			message,
+		),
+	)
 
-		_, err = UnmarshalMessage(
+	DescribeTable(
+		"execution of UnmarshalMessage errodes",
```

Comment:
typo `errodes` -> `errors`?

Also, I don't think there's any need to duplicate each of these tests for JSON/PB when the error is related to the content type itself -- this will bloat quickly when we add new encoding types.